### PR TITLE
Buildnummer komplett entfernt. …

### PR DESCRIPTION
**Pull request by alex1702**
Es wird jetzt die Versionsnummer als Buildnummer ausgegeben.
Hardgecodete Versionsnummer für obsolet erklärt, da sie nun aus der version.properties geladen wird.
Betrifft Issue #86

### CODE REVIEW

File: src/main/java/mSearch/Const.java [hunk 1]
```diff
@@ -23,7 +23,7 @@
 
 public class Const {
 
-    public static final String VERSION = "13";
+    public static final String VERSION = "13"; //Obsolet
```
Comment (Nicklas2751) [inline src/main/java/mSearch/Const.java]: Anstatt `//Obsolet` schreibe lieber `@Deprecated`davor.

File: src/main/java/mSearch/tool/Functions.java [hunk 2]
```diff
@@ -159,15 +159,12 @@ public static String getCompileDate() {
 
     public static String getBuildNr() {
         final ResourceBundle rb;
-        String TOKEN_BUILD = "BUILD";
         String TOKEN_VERSION = "VERSION";
-        String msg = "0-0";
+        String msg = "-1";
```
Comment (Nicklas2751) [inline src/main/java/mSearch/tool/Functions.java]: Wieso -1? O.O

Comment (alex1702) [inline src/main/java/mSearch/tool/Functions.java]: Auffälliger, dachte ich mir so dabei.

File: src/main/java/mSearch/Const.java [hunk 3]
```diff
@@ -23,7 +23,7 @@
 
 public class Const {
 
-    public static final String VERSION = "13";
+    @Deprecated public static final String VERSION = "13";
```
Comment (alex1702) [inline src/main/java/mSearch/Const.java]: MAJOR Add the missing @deprecated Javadoc tag. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)
INFO Do not forget to remove this deprecated code someday. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/mSearch/tool/Functions.java [hunk 4]
```diff
@@ -157,23 +157,35 @@ public static String getCompileDate() {
...
+                return new Version(rb.getString(TOKEN_VERSION));
+            }
+        } catch (Exception e) {
+            Log.errorLog(134679898, e);
+        }
+        return new Version("");
+    }
+    
+    @Deprecated
     public static String getBuildNr() {
```
Comment (alex1702) [inline src/main/java/mSearch/tool/Functions.java]: MAJOR Add the missing @deprecated Javadoc tag. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)
INFO Do not forget to remove this deprecated code someday. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/mSearch/tool/Version.java [hunk 5]
```diff
@@ -0,0 +1,92 @@
+package mSearch.tool;
```
Comment (alex1702) [inline src/main/java/mSearch/tool/Version.java]: MINOR Rename this package name to match the regular expression '^[a-z]+(\.[a-z][a-z0-9]*)*$'. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS00120)

File: src/main/java/mSearch/tool/Version.java [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+	 */
+	public int toNumber() {
+		return major * 100 + minor * 10 + patch;
+	}
+	
+	/**
+	 * Gibt die Version als String zurück
+	 * @return String mit der Version
+	 */
+	public String toString() {
```
Comment (alex1702) [inline src/main/java/mSearch/tool/Version.java]: MAJOR Add the "@Override" annotation above this method signature [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1161)

**Comment by alex1702** [comment]
SonarQube analysis reported 7 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 3 major
* ![MINOR][MINOR] 1 minor
* ![INFO][INFO] 2 info

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [Functions.java#L150](https://github.com/mediathekview/MSearch/blob/aaf28548283d16b71d2a141e2eac72ac8209b706/src/main/java/mSearch/tool/Functions.java#L150): Define a constant instead of duplicating this literal "version" 3 times. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1192)

[CRITICAL]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Comment by alex1702** [comment]
SonarQube analysis reported 6 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 1 minor
* ![INFO][INFO] 2 info

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [Functions.java#L150](https://github.com/mediathekview/MSearch/blob/a58ac29438ad71f7045503ba9acf228223b55825/src/main/java/mSearch/tool/Functions.java#L150): Define a constant instead of duplicating this literal "version" 3 times. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1192)

[CRITICAL]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Comment by alex1702** [comment]
SonarQube analysis reported 5 issues
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 1 minor
* ![INFO][INFO] 2 info

Watch the comments in this conversation to review them.

[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Pull request merged**